src/lib/shadow-shim-helper-rs/src/shim_shmem.rs

```diff
@@ -434,7 +438,9 @@ impl Default for SiginfoWrapper {
 
 impl From<libc::siginfo_t> for SiginfoWrapper {
     fn from(s: libc::siginfo_t) -> Self {
-        Self(s)
+        static_assertions::assert_eq_align!(SiginfoWrapper, libc::siginfo_t);
+        static_assertions::assert_eq_size!(SiginfoWrapper, libc::siginfo_t);
+        Self(unsafe { core::mem::transmute(s) })
     }
 }
```

Comment:
Should make sure there's no padding bytes in `libc::siginfo_t` so that it doesn't transmute uninitialized bytes into initialized fields.

Reply:
Removed. I got rid of most of the conversions to/from the libc type; I don't think there's any guarantee that it'll always be compatible with the kernel definition.

There's still one place in the shim where we unsafely assume this by casting the pointer; I added a TODO to avoid it by installing our native signal handlers without going through libc, so that our handler actually gets `linux_siginfo_t`.